narwhal/primary/src/proposer.rs

```diff
@@ -512,33 +515,38 @@ impl Proposer {
 
                 Some((commit_round, commit_headers)) = self.rx_committed_own_headers.recv() => {
                     // Remove committed headers from the list of pending
-                    for round in &commit_headers {
-                        self.proposed_headers.remove(round);
+                    let mut max_committed_round = 0;
+                    for round in commit_headers {
+                        max_committed_round = max_committed_round.max(round);
+                        let Some(_) = self.proposed_headers.remove(&round) else {
+                            info!("Own committed header not found at round {round}, probably because of restarts.");
+                            continue;
```

Comment:
What's the intention of the `continue` here? IIUC the intention is that we want to skip processing any other committed headers as we have probably restarted - in this case you want to tag the  loop outside of `tokio::select` and do `continue` on that one.